scene/resources/navigation_mesh_source_geometry_data_3d.cpp

```diff
@@ -165,6 +165,17 @@ void NavigationMeshSourceGeometryData3D::add_faces(const PackedVector3Array &p_f
 	_add_faces(p_faces, root_node_transform * p_xform);
 }
 
+void NavigationMeshSourceGeometryData3D::merge(const Ref<NavigationMeshSourceGeometryData3D> &p_other_geometry) {
+	// No need to worry about `root_node_transform` here as the vertices are already xformed.
+	Vector<float>::Size number_of_vertices_before_merge = vertices.size();
+	Vector<int>::Size number_of_indices_before_merge = indices.size();
+	vertices.append_array(p_other_geometry->vertices);
+	indices.append_array(p_other_geometry->indices);
+	for (Vector<int>::Size i = number_of_indices_before_merge; i < indices.size(); i++) {
+		indices.set(i, indices[i] + number_of_vertices_before_merge / 3);
+	}
```

Comment:
The `Vector<T>::Size` was intended to be an implementation detail IMO in #86730. Since in the end it was no longer platform specific and is always 
```cpp
        typedef int64_t Size;
        typedef uint64_t USize;
```
Maybe we should remove the typedef and just use `int64_t` and `uint64_t`, to avoid this implementation detail spilling into the whole codebase?

CC @reduz @bruvzg

Reply:
I've changed it to `int64_t` to not block this PR